src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -1502,9 +1502,21 @@ window.qBittorrent.DynamicTable ??= (() => {
             const rows = this.rows.getValues();
             const useRegex = $("torrentsFilterRegexBox").checked;
             const filterText = $("torrentsFilterInput").value.trim().toLowerCase();
-            const filterTerms = (filterText.length > 0)
-                ? (useRegex ? new RegExp(filterText) : filterText.split(" "))
-                : null;
+            let filterTerms = null;
```

Comment:
Could be done in one line ofc, but I thought at that point readability was taking a hit

```javascript
      try {
            const filterTerms = (filterText.length > 0)
                  ? (useRegex ? new RegExp(filterText) : filterText.split(" "))
                  : null;
      }
      catch (e) {
            console.error(`Invalid regex pattern: ${filterText}`);
            return filteredRows;
      }
```

maybe I'm overthinking it tho

Reply:
I think the code piece above is fine, you can use it if you want.

IMO the `console.error` is a bit excessive, I would suggest removing it either way.